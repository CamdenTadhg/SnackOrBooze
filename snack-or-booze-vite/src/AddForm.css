.AddForm-form label{
    margin-right: 1%;
}

.AddForm-form input{
    width: 75%;
}