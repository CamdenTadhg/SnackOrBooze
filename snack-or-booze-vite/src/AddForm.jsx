import React, {useState} from 'react';
import {Card, CardBody, CardTitle} from "reactstrap";
import SnackOrBoozeApi from "./Api";
import './AddForm.css'
import 'bootstrap/dist/css/bootstrap.min.css';

//Component renders form to create new snacks and drinks in the database
function AddForm({setSnacks, setDrinks}) {
    //set initial state for form
    const initialState = {
        type: 'snacks',
        id: '',
        name: '',
        description: '',
        recipe: '',
        serve: ''
    }
    const [formData, setFormData] = useState(initialState);
    //handle change to form entries
    const handleChange = (event) => {
        const {name, value} = event.target;
        setFormData(fData => ({
            ...fData,
            [name]: value
        }));
    };
    
    //handle form submission. Create a new snack or drink and get the updated list
    const handleSubmit = async (event) => {
        event.preventDefault();
        const {type, id, name, description, recipe, serve} = formData;
        const data = {id, name, description, recipe, serve};
        await SnackOrBoozeApi.addGoodies(type, data);
        setFormData(initialState);
        let snacks = await SnackOrBoozeApi.getGoodies('snacks');
        let drinks = await SnackOrBoozeApi.getGoodies('drinks');
        setSnacks(snacks);
        setDrinks(drinks);
    }

    return (
      <section className="col-md-4">
        <Card>
          <CardBody>
            <CardTitle className="font-weight-bold text-center">
             Add A Menu Item
            </CardTitle>
            <form onSubmit={handleSubmit} className='AddForm-form'>
                <div>
                    <label htmlFor="type">Type:</label>
                    <select id="type" name="type" value={formData.type} onChange={handleChange}>
                        <option value="snacks">Snack</option>
                        <option value="drinks">Drink</option>
                    </select>
                </div>
                <div>
                    <label htmlFor="id">Id:  </label>
                    <input type='text' placeholder='id' id="id" name='id' value={formData.id} onChange={handleChange}/>
                </div>
                <div>
                    <label htmlFor="name">Name:  </label>
                    <input type='text' placeholder='name' id="name" name='name' value={formData.name} onChange={handleChange} />
                </div>
                <div>
                    <label htmlFor="description">Description: </label>
                    <input type='text' placeholder='description' id="description" name='description' value={formData.description} onChange={handleChange} />
                </div>
                <div>
                    <label htmlFor="recipe">Recipe: </label>
                    <input type='text' placeholder='recipe' id="recipe" name='recipe' value={formData.recipe} onChange={handleChange} />
                </div>
                <div>
                    <label htmlFor="serve">Serve:</label> 
                    <input type='text' placeholder='serve' id="serve" name='serve' value={formData.serve} onChange={handleChange} />
                </div> 
                <button>Add</button>
            </form>
          </CardBody>
        </Card>
      </section>
    );
  }

export default AddForm;