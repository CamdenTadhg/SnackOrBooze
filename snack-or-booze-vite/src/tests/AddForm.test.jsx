import {render} from '@testing-library/react';
import {test, expect} from 'vitest';
import AddForm from '../AddForm';

test('renders the AddForm component', () => {
    render(<AddForm/>);
});

test('matches snapshot', function() {
    const addForm = render(<AddForm />);
    expect(addForm).toMatchSnapshot();
});