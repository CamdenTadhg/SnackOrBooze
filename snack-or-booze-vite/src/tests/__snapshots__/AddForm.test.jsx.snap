// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="col-md-4"
      >
        <div
          class="card"
        >
          <div
            class="card-body"
          >
            <div
              class="font-weight-bold text-center card-title"
            >
              Add A Menu Item
            </div>
            <form
              class="AddForm-form"
            >
              <div>
                <label
                  for="type"
                >
                  Type:
                </label>
                <select
                  id="type"
                  name="type"
                >
                  <option
                    value="snacks"
                  >
                    Snack
                  </option>
                  <option
                    value="drinks"
                  >
                    Drink
                  </option>
                </select>
              </div>
              <div>
                <label
                  for="id"
                >
                  Id:  
                </label>
                <input
                  id="id"
                  name="id"
                  placeholder="id"
                  type="text"
                  value=""
                />
              </div>
              <div>
                <label
                  for="name"
                >
                  Name:  
                </label>
                <input
                  id="name"
                  name="name"
                  placeholder="name"
                  type="text"
                  value=""
                />
              </div>
              <div>
                <label
                  for="description"
                >
                  Description: 
                </label>
                <input
                  id="description"
                  name="description"
                  placeholder="description"
                  type="text"
                  value=""
                />
              </div>
              <div>
                <label
                  for="recipe"
                >
                  Recipe: 
                </label>
                <input
                  id="recipe"
                  name="recipe"
                  placeholder="recipe"
                  type="text"
                  value=""
                />
              </div>
              <div>
                <label
                  for="serve"
                >
                  Serve:
                </label>
                <input
                  id="serve"
                  name="serve"
                  placeholder="serve"
                  type="text"
                  value=""
                />
              </div>
              <button>
                Add
              </button>
            </form>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="col-md-4"
    >
      <div
        class="card"
      >
        <div
          class="card-body"
        >
          <div
            class="font-weight-bold text-center card-title"
          >
            Add A Menu Item
          </div>
          <form
            class="AddForm-form"
          >
            <div>
              <label
                for="type"
              >
                Type:
              </label>
              <select
                id="type"
                name="type"
              >
                <option
                  value="snacks"
                >
                  Snack
                </option>
                <option
                  value="drinks"
                >
                  Drink
                </option>
              </select>
            </div>
            <div>
              <label
                for="id"
              >
                Id:  
              </label>
              <input
                id="id"
                name="id"
                placeholder="id"
                type="text"
                value=""
              />
            </div>
            <div>
              <label
                for="name"
              >
                Name:  
              </label>
              <input
                id="name"
                name="name"
                placeholder="name"
                type="text"
                value=""
              />
            </div>
            <div>
              <label
                for="description"
              >
                Description: 
              </label>
              <input
                id="description"
                name="description"
                placeholder="description"
                type="text"
                value=""
              />
            </div>
            <div>
              <label
                for="recipe"
              >
                Recipe: 
              </label>
              <input
                id="recipe"
                name="recipe"
                placeholder="recipe"
                type="text"
                value=""
              />
            </div>
            <div>
              <label
                for="serve"
              >
                Serve:
              </label>
              <input
                id="serve"
                name="serve"
                placeholder="serve"
                type="text"
                value=""
              />
            </div>
            <button>
              Add
            </button>
          </form>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
